src/browser-polyfill.js

```diff
@@ -6,7 +6,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 "use strict";
 
-if (typeof browser === "undefined") {
+if (typeof browser === "undefined" || browser instanceof window.Node) {
```

Comment:
Shouldn't it be `window.browser instanceof window.Node` to avoid an exception in strict mode?

Reply:
Change to:

```javascript
if (typeof browser !== "object" || browser && Object.getPrototypeOf(browser) !== Object.prototype) {`
```

or (if you prefer shorter code or don't want to rely on `Object.getPrototypeOf`):

```javascript
if (typeof browser === "undefined" || browser && browser.__proto__ !== Object.prototype) {
```

or (if you don't want to rely on `Object`):

```javascript
if (typeof browser === "undefined" || browser && browser.__proto__ !== ({}).__proto__) {
```

Why? Well:
1. Using `instanceof window.Node` won't rule out tampering. You'll see that in this case, `x` is a `HTMLCollection`: `data:text/html,<a id=x href>a</a><a id=x href>b</a><script>console.log(x)</script>`
2. If someone has happened to define a primitive value as "browser", then we still want to overwrite it. (this is an accidental side-effect of my chosen implementation)

See also https://www.w3.org/TR/html5/browsers.html#named-access-on-the-window-object